### PR TITLE
Write cpp package

### LINKED ISSUE #3: Write C++ package
We would like to have a C++ library that does not have Matlab license dependencies.

### PR DESCRIPTION
**Pull request by JakobThumm**
closes #3 
This PR adds the open-source C++ library for the LongTermPlanner.

### CODE REVIEW

**Review by yannickBurkhardt — COMMENTED** [review]
https://github.com/yannickBurkhardt/LongTermPlanner/tree/3-write-cpp-package/Testing/Temporary

Can this folder be removed?

File: tests/unittests/testTimeScaling.m [hunk 1]
```diff
@@ -31,7 +31,7 @@
 % Test all scenarios
 success = 0;
 fail = 0;
-for i=1:num_scenarios
+for i=12:num_scenarios
```
Comment (yannickBurkhardt) [inline tests/unittests/testTimeScaling.m]: Is there a reason why you skip the first 11 scenarios or is this change a typo?

**Review by yannickBurkhardt — COMMENTED** [review]
https://github.com/yannickBurkhardt/LongTermPlanner/blob/3-write-cpp-package/tests/debug_t_rel_issue.m

Can this file be removed?

File: tests/unittests/testTimeScaling.m (hunk 1 above)
Comment (JakobThumm) [reply tests/unittests/testTimeScaling.m]: I reverted the change.

**Review by yannickBurkhardt — APPROVED** [review]

**Pull request merged**